posts/2023-10-23-new-diagnostic-analyzers-feature.md

```diff
@@ -0,0 +1,56 @@
+---
+title: "A closer look: Diagnostic Analyzers with Roslyn"
+author: joreyk
+popular: false
+image: https://www.jenx.si/wp-content/uploads/2020/10/cs-roslyn-logo.png
+tags: ['.NET']
+---
+
+Let's take a closer look at the DiagnosticAnalyzer added in Stride.Core.CompilerServices
+
+---
+
+Table of Contents:
+
+[[TOC]]
+
+Stride advances further in utility and features every week.
+This blog will cover the new `DiagnosticAnalyzer` features in `Stride.Core.CompilerServices`
+
+## What is a DiagnosticAnalyzer?
+
+A `DiagnosticAnalyzer` is a Roslyn feature to scan your code while you are typing in your IDE.
+C# uses them too, everytime you see a warning or red squiggly lines in your IDE, a `DiagnosticAanalyzer` get's triggered.
+This analyzation has `NOTHING` to do with telemetry.
+
+## What does it do?
+
+The new `Diagnostics` cover mostly `Serialization`, the analyzers will create warnings when code is written that is incompatible with [Strides Serialization Rules of Thumb](https://github.com/stride3d/stride-docs/blob/master/en/manual/scripts/serialization.md#rule-of-thumb).
```

Comment:
Did you want to link actually to this location 

https://doc.stride3d.net/latest/en/manual/scripts/serialization.html

Reply:
No problem. And obviously, I remember as I mess around docs 🤣